swupd/packs.go

```diff
@@ -115,15 +135,15 @@ func WritePack(w io.Writer, m *Manifest, fromVersion uint32, outputDir, chrootDi
 
 	var fullChrootDir string
 	if chrootDir != "" {
-		fullChrootDir = filepath.Join(chrootDir, fmt.Sprint(m.Header.Version), "full")
+		fullChrootDir = filepath.Join(chrootDir, fmt.Sprint(toVersion), "full")
 	}
 
 	done := make(map[Hashval]bool)
-	for i, f := range m.Files {
+	for i, f := range toManifest.Files {
 		entry := &info.Entries[i]
 		entry.File = f
 
-		if f.Version <= fromVersion {
+		if f.Version <= fromVersion || hashesInFrom[f.Hash] {
```

Comment:
`fromVersion` will not be set if `fromManifest == nil`

Reply:
For zero packs, fromVersion is zero, and that works for this condition because the f.Version will always be greater than zero.

Reply:
oh, right, because zero values. +1, thanks for the comment.